rtcpeerconnection.go

```diff
@@ -484,18 +517,148 @@ func (pc *RTCPeerConnection) CreateAnswer(options *RTCAnswerOptions) (RTCSession
 
 	d = d.WithValueAttribute(sdp.AttrKeyGroup, bundleValue)
 
-	pc.CurrentLocalDescription = &RTCSessionDescription{
+	desc := RTCSessionDescription{
 		Type:   RTCSdpTypeAnswer,
 		Sdp:    d.Marshal(),
 		parsed: d,
 	}
-	return *pc.CurrentLocalDescription, nil
+	pc.lastAnswer = desc.Sdp
+
+	// FIXME: This doesn't follow the JS API spec, but removing it
+	// would mean our examples and existing code have to change
+	if err := pc.SetLocalDescription(desc); err != nil {
+		return RTCSessionDescription{}, err
+	}
+	return desc, nil
 }
 
-// // SetLocalDescription sets the SessionDescription of the local peer
-// func (pc *RTCPeerConnection) SetLocalDescription() {
-// 	panic("not implemented yet") // FIXME NOT-IMPLEMENTED nolint
-// }
+// 4.4.1.6 Set the RTCSessionDescription
+func (pc *RTCPeerConnection) setDescription(sd *RTCSessionDescription, op rtcStateChangeOp) error {
+	if pc.isClosed {
+		return &rtcerr.InvalidStateError{Err: ErrConnectionClosed}
+	}
+
+	cur := pc.SignalingState
+	setLocal := rtcStateChangeOpSetLocal
+	setRemote := rtcStateChangeOpSetRemote
+	newSdpDoesNotMatchOffer := &rtcerr.InvalidModificationError{Err: errors.New("New sdp does not match previous offer")}
+	newSdpDoesNotMatchAnswer := &rtcerr.InvalidModificationError{Err: errors.New("New sdp does not match previous answer")}
+
+	var nextState RTCSignalingState
+	var err error
+	switch op {
+	case setLocal:
```

Comment:
Oh hell yes, thanks for adding all this error handling around signaling!

We have gotten so many issues around this, nice to fix them up.